src/sentry/static/sentry/app/components/events/interfaces/spans/utils.tsx

```diff
@@ -236,29 +237,29 @@ export const getHumanDuration = (duration: number): string => {
   return `${durationMS.toFixed(2)}ms`;
 };
 
-const COLORS = [
-  '#7274AC',
-  '#9D85B8',
-  '#BF8CB6',
-  '#CF7CA0',
-  '#ED8898',
-  '#F6A189',
-  '#F8B26D',
-  '#F7D36E',
-
-  // reverse fade
-
-  '#F8B26D',
-  '#F6A189',
-  '#ED8898',
-  '#CF7CA0',
-  '#BF8CB6',
-  '#9D85B8',
-];
-export const pickSpanBarColour = (input: number): string => {
-  const index = input % COLORS.length;
-
-  return COLORS[index];
+const getLetterIndex = (letter: string): number => {
+  const index = 'abcdefghijklmnopqrstuvwxyz'.indexOf(letter) || 0;
+  return index === -1 ? 0 : index;
+};
+
+export const pickSpanBarColour = (input: string | undefined): string => {
+  // We pick the color for span bars using the first two letters of the op name.
+  // That way colors stay consistent between transactions.
+
+  if (!input || input.length < 2) {
+    return divergentColorScale.blue;
+  }
+  if (spanColors[input]) {
+    return spanColors[input];
+  }
+
+  const colorsAsArray = Object.keys(divergentColorScale).map(
```

Comment:
@Chrissy I recommend moving the variable, `colorsAsArray `, outside of the function `pickSpanBarColour`; so that it doesn't need to be reconstructed whenever the function `pickSpanBarColour` is called.